src/hooks/useReportError.ts

```diff
@@ -15,18 +15,17 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { useEffect } from "react";
 import reportError from "@/telemetry/reportError";
+import { usePreviousValue } from "@/hooks/usePreviousValue";
 
 /**
- * React hook to report an error once.
+ * React hook to report an error if it's different from the previous error.
  */
 function useReportError(error: unknown): void {
-  useEffect(() => {
-    if (error) {
-      reportError(error);
-    }
-  }, [error]);
+  const previousError = usePreviousValue(error);
+  if (error && error !== previousError) {
+    reportError(error);
+  }
```

Comment:
I don't like this, it's repeating the logic in every instance and the hook still uses useEffect inside so it's not actually changing its performance as the slack message suggested. 

I like using new hooks to incapsulate complexity, but this just moves it out of React's patterns.

What you're most likely looking for is a hook exactly like useEffect, but which also includes the condition.

I was actually looking to add an error-specific hook here, like:

```suggestion
  useReportErrorOnce(error);
```

And the others might just be:

```js
useEffectIfTruthy(error, () => reportError(error))
```

However neither `useEffectIfTruthy` nor `usePreviousValue` lead to code that is faster, safer nor substantially more readable.

Reply:
@fregante see previous discussion here: https://github.com/pixiebrix/pixiebrix-extension/pull/7852#discussion_r1517185204

@grahamlangford @fungairino I think useEffect vs. usePreviousValue for firing events/errors would be a good one for a time-boxed discussion/vote so we can decide and the move on